client/components/Cart.js

```diff
@@ -1,109 +1,106 @@
-import React from "react";
-import { useState, useEffect } from "react";
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import { connect } from "react-redux";
+import {
+  addToCart,
+  removeFromCart,
+  decreaseItem,
+  clearCart,
+} from "../store/cart";
 //import DeleteIcon from "@mui/icons-material/Delete";
 //import AddCircleRoundedIcon from "@mui/icons-material/AddCircleRounded";
 
-const cartFromLocalStorage = JSON.parse(
-  window.localStorage.getItem("cart") || "[]"
-);
+export class Cart extends Component {
+  constructor(props) {
+    super(props);
+  }
 
-export default function Cart(props) {
-  const [cart, setCart] = useState(cartFromLocalStorage);
-
-  useEffect(() => {
-    window.localStorage.setItem("cart", JSON.stringify(cart));
-  }, [cart]);
-  const itemsPrice = cart.reduce((a, c) => a + c.price * c.qty, 0);
-  const tax = itemsPrice * 0.08;
-  const totalPrice = itemsPrice + tax;
-  const onAdd = (product) => {
-    const exist = cart.find((item) => item.id === product.id);
-    if (exist) {
-      setCart(
-        cart.map((item) =>
-          item.id === product.id ? { ...exist, qty: exist.qty + 1 } : item
-        )
-      );
-    } else {
-      setCart([...cart, { ...product, qty: 1 }]);
-    }
-  };
-  const onRemove = (product) => {
-    const exist = cart.find((item) => item.id === product.id);
-    if (exist.qty === 1) {
-      setCart(cart.filter((item) => item.id !== product.id));
-    } else {
-      setCart(
-        cart.map((item) =>
-          item.id === product.id ? { ...exist, qty: exist.qty - 1 } : item
-        )
-      );
-    }
-  };
-
-  return (
-    <div>
-      <h3>Cart Items</h3>
-      <div>{cart.length === 0 && <div>Cart is Empty</div>}</div>
-      {cart.map((item) => {
-        return (
-          <div key={item.id}>
-            <img src={item.imageURL}></img>
-            <div>{item.name}</div>
+  render() {
+    const { cart } = this.props;
+    const itemsPrice = cart.reduce((a, c) => a + c.price * c.count, 0);
```

Comment:
looks at entire cart array and calculates total of all cart items * count of item in the cart